controllers/barbicankeystonelistener_controller.go

```diff
@@ -401,6 +402,36 @@ func (r *BarbicanKeystoneListenerReconciler) reconcileNormal(ctx context.Context
 
 	Log.Info(fmt.Sprintf("[KeystoneListener] Got secrets '%s'", instance.Name))
 
+	// check for  ApplicationCredential
+	acName := fmt.Sprintf("ac-%s", barbican.ServiceName)
+	ac := &keystonev1.ApplicationCredential{}
+	err = r.Client.Get(ctx, types.NamespacedName{Namespace: instance.Namespace, Name: acName}, ac)
+	if k8s_errors.IsNotFound(err) {
+		Log.Info("No ApplicationCredential CR found, using password auth", "ac", acName)
+	} else if err != nil {
+		return ctrl.Result{}, err
+	} else {
+		secretName := ac.Status.SecretName
+		if secretName == "" {
+			Log.Info("ApplicationCredential created but Secret not yet ready, requeueing", "ac", acName)
+			return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
+		}
+		hashAC, res, err := secret.VerifySecret(
+			ctx,
+			types.NamespacedName{Name: secretName, Namespace: instance.Namespace},
+			[]string{"AC_ID", "AC_SECRET"},
+			helper.GetClient(),
+			10*time.Second,
+		)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		if res.RequeueAfter > 0 {
+			return res, nil
+		}
+		configVars["secret-"+secretName] = env.SetValue(hashAC)
+	}
+
```

Comment:
Same comments as above, but actually, we should think about whether we need this logic at all.  I don't think that the barbican-keystone-listener actually talks to keystone or uses the keystone creds.  In which case, maybe we don't need to verify these parameters at all.